### PR TITLE
Break when minY >= bottom

### LINKED ISSUE #1105: CloneAs fails with dimensions 374x209 and MaxDegreeOfParallelism=16
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp (ImageSharp@v1.0.0-unstable0542)
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
For some pixel data loaded images, converting from Bgra32 to Rgba32 fails. Saving the Bgra32 image to a PNG succeeds, and the data is valid / non-corrupt.

### Steps to Reproduce
Use repo below
[Test.zip](https://github.com/SixLabors/ImageSharp/files/4145656/Test.zip)

### System Configuration
Windows 10 Pro x64
Docker / aspnet runtime 3.1

- ImageSharp version: v1.0.0-unstable0542
- Other ImageSharp packages and versions: --
- Environment (Operating system, version and so on): Windows 10 Pro x64
- .NET Framework version: NetCoreApp 3.1
- Additional information:
Hello World

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
The `ParallelHelper` calculation did not correctly handle certain edge cases when the height of a vertical step exceeded the height/number of steps resulting in `RowInterval.MinY` greater than `RowInterval.MaxY`.

This fix returns early in the parallel action when this case is detected. Fixes #1105

### CODE REVIEW

File: src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs [hunk 1]
```diff
@@ -62,14 +62,23 @@ public static void IterateRows(Rectangle rectangle, Configuration configuration,
...
                 parallelOptions,
                 i =>
                     {
-                        int yMin = rectangle.Top + (i * verticalStep);
-                        int yMax = Math.Min(yMin + verticalStep, rectangle.Bottom);
+                        int yMin = top + (i * verticalStep);
+
+                        if (yMin >= bottom)
+                        {
+                            return;
```
Comment (antonfirsov) [inline src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs]: Note to ourselves:
we launch a few threads out to the void, but it doesn't worth to bother more with this special case.

File: tests/ImageSharp.Tests/Helpers/ParallelHelperTests.cs [hunk 2]
```diff
@@ -39,7 +39,10 @@ public ParallelHelperTests(ITestOutputHelper output)
                     { 4, 123, 323, 50, 50, 4 },
                     { 4, 0, 1201, 301, 298, 4 },
                     { 8, 10, 236, 29, 23, 8 },
-                    { 16, 0, 209, 14, 13, 15 }
+                    { 16, 0, 209, 14, 13, 15 },
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Helpers/ParallelHelperTests.cs]: Ace. Was hoping you'd add more 👍

File: src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs]: Yeah, I did notice this. That's why I was hoping you were gonna add more tests, it took me a while to figure it all out. 😄
